lib/ansible/plugins/connection/aws_ssm.py

```diff
@@ -0,0 +1,467 @@
+# Based on the ssh connection plugin by Michael DeHaan
+#
+# Copyright: (c) 2018, Pat Sharkey <psharkey@cleo.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+author:
+- Pat Sharkey (@psharkey) <psharkey@cleo.com>
+- HanumanthaRao MVL (@hanumantharaomvl) <hanumanth@flux7.com>
+- Gaurav Ashtikar (@gau1991 )<gaurav.ashtikar@flux7.com>
+connection: aws_ssm
+short_description: execute via AWS Systems Manager
+description:
+- This connection plugin allows ansible to execute tasks on an EC2 instance via the aws ssm CLI.
+version_added: "2.8"
+requirements:
+- The remote EC2 instance must be running the AWS Systems Manager Agent (SSM Agent).
+- The control machine must have the aws session manager plugin installed.
+- The remote EC2 instance must have the curl installed.
+options:
+  instance_id:
+    description: The EC2 instance ID.
+    vars:
+    - name: ansible_aws_ssm_instance_id
+  region:
+    description: The region the EC2 instance is located.
+    vars:
+    - name: ansible_aws_ssm_region
+    default: 'us-east-1'
+  bucket_name:
+    description: The name of the S3 bucket used for file transfers.
+    vars:
+    - name: ansible_aws_ssm_bucket_name
+  plugin:
+    description: This defines the location of the session-manager-plugin binary.
+    vars:
+    - name: ansible_aws_ssm_plugin
+    default: '/usr/local/bin/session-manager-plugin'
+  retries:
+    description: Number of attempts to connect.
+    default: 3
+    type: integer
+    vars:
+    - name: ansible_aws_ssm_retries
+  timeout:
+    description: Connection timeout seconds.
+    default: 60
+    type: integer
+    vars:
+    - name: ansible_ssm_timeout
+"""
+
+EXAMPLES = '''
+
+# Stop Spooler Process on Windows Instances
+- name: Stop Spooler Service on Windows Instances
+  vars:
+    ansible_connection: aws_ssm
+    ansible_shell_type: powershell
+    ansible_aws_ssm_bucket_name: nameofthebucket
+    ansible_aws_ssm_region: us-east-1
+  tasks:
+    - name: Stop spooler service
+      win_service:
+        name: spooler
+        state: stopped
+
+# Install a Nginx Package on Linux Instance
+- name: Install a Nginx Package
+  vars:
+    ansible_connection: aws_ssm
+    ansible_aws_ssm_bucket_name: nameofthebucket
+    ansible_aws_ssm_region: us-west-2
+  tasks:
+    - name: Install a Nginx Package
+      yum:
+        name: nginx
+        state: present
+
+# Create a directory in Windows Instances
+- name: Create a directory in Windows Instance
+  vars:
+    ansible_connection: aws_ssm
+    ansible_shell_type: powershell
+    ansible_aws_ssm_bucket_name: nameofthebucket
+    ansible_aws_ssm_region: us-east-1
+  tasks:
+    - name: Create a Directory
+      win_file:
+        path: "C:\\Windows\\temp"
```

Comment:
No need to quote the path, not using double quotes also means you don't need to double up on `\`, e.g. just do `path: C:\Windows\temp`.

Reply:
HI @jborean93 , I would believe we need that due to pycodestyle / W605 else we would end up with  - ["aws_ssm.py:93:18: W605 invalid escape sequence '\\W'"]}

Reply:
Add the `r'''` to https://github.com/ansible/ansible/pull/49652/files/6dc61c94e3abd3cf74998b78f4fa08956a95c269#diff-1c417562a0944ee1362da84b93d9695dR56 to avoid it. We do this for all our module examples so we don't have to worry about escaping these values.

Reply:
HI @jborean93, That was awesome and have updated in https://github.com/ansible/ansible/pull/49652/commits/b9b7cfa8cc6f9a1bf0182cbc4d23c6fc34cbefcb .  Appreciate your flash responses.